Elevel/Application/Features/TestCommands/CreateTestCommand.cs

```diff
@@ -157,48 +163,118 @@ private void CreateTestQuestionsForGrammar(List<Guid> questionIds, Test test)
             /// <returns>true if not created, false if created</returns>
-            private async Task<bool> CreateTestGrammarQuestionsAsync(Test test, CancellationToken cancelationtoken)
+            private async Task<List<TestQuestion>> CreateTestGrammarQuestionsAsync(Test test, CancellationToken cancelationtoken)
             {
                 var questions = await GetQuestionListAsync(test.Level);
                 if (questions.Count() < GRAMMAR_TEST_COUNT)
                 {
-                    return true;
+                    throw new ValidationException("Not Enough questions");
                 }
 
                 var questionIds = GetQuestionIds(questions, GRAMMAR_TEST_COUNT);
 
-                CreateTestQuestionsForGrammar(questionIds, test);
+                var testQuestions = CreateTestQuestions(questionIds, test);
 
-                return false;
+                return testQuestions;
             }
             /// <summary>
             /// Generates Audition test questions for received test
             /// </summary>
             /// <param name="test"></param>
             /// <param name="cancelationtoken"></param>
             /// <returns></returns>
-            private async Task<bool> CreateTestAuditionQuestionsAsync(Test test, CancellationToken cancelationtoken)
+            private async Task<List<TestQuestion>> CreateTestAuditionQuestionsAsync(Test test, CancellationToken cancelationtoken)
             {
                 var questions = await GetQuestionListAsync(test.Level, test.AuditionId);
 
                 if (questions.Count() < AUDITION_TEST_COUNT)
                 {
-                    return true;
+                    throw new ValidationException("Not Enough questions");
                 }
 
                 var questionIds = GetQuestionIds(questions, AUDITION_TEST_COUNT);
 
-                CreateTestQuestionsForGrammar(questionIds, test);
+                var testQuestions = CreateTestQuestions(questionIds, test);
+
+                return testQuestions;
+            }
+
+            private async Task<IQueryable<Question>> GetQuestionsByAuditionIdAsync(IEnumerable<TestQuestion> testQuestions, Guid? auditionId)
+            {
+                var questions = new List<Question>();
+                foreach (var testQuestion in testQuestions)
+                {
+                    var question = await _context.Questions.FirstOrDefaultAsync(x => x.Id == testQuestion.QuestionId).ConfigureAwait(false);
+                    if (question.AuditionId == auditionId)
+                    {
+                        questions.Add(question);
+                    }
+                }
+                return questions.AsQueryable();
+            }
+            private async Task AddAnswerAsync(List<QuestionDto> questions)
+            {
+                foreach (var question in questions)
+                {
+                    question.Answers = await GetAnswerDtosAsync(question.Id).ConfigureAwait(false);
```

Comment:
database queries in a loop it's a bad practice. You should find solution with one query to the DB.